lib/HLSL/HLOperationLower.cpp

```diff
@@ -741,17 +740,16 @@ Value *TranslateD3DColorToUByte4(CallInst *CI, IntrinsicOp IOP,
 
   if (Ty->isVectorTy()) {
     static constexpr int supportedVecElemCount = 4;
-    if (Ty->getVectorNumElements() == supportedVecElemCount) {
-      toByteConst =
-          ConstantVector::getSplat(supportedVecElemCount, toByteConst);
-      // Swizzle the input val -> val.zyxw
-      std::vector<int> mask{2, 1, 0, 3};
-      val = Builder.CreateShuffleVector(val, val, mask);
-    } else {
+    if (Ty->getVectorNumElements() != supportedVecElemCount) {
       llvm_unreachable(
           "Unsupported input type for intrinsic D3DColorToUByte4.");
       return UndefValue::get(CI->getType());
     }
+
+    toByteConst = ConstantVector::getSplat(supportedVecElemCount, toByteConst);
+    // Swizzle the input val -> val.zyxw
+    std::vector<int> mask{2, 1, 0, 3};
+    val = Builder.CreateShuffleVector(val, val, mask);
```

Comment:
Maybe going too far here......but the unnecessary heap allocation makes me sad....

I think this would work - I think it should be hitting the `ArrayRef` overload of `CreateShuffleVector`?
```suggestion
    const int mask[] = {2, 1, 0, 3};
    val = Builder.CreateShuffleVector(val, val, mask);
```

Feel free to ignore if you think this is too churny.